framework/CodeInterfaces/RAVEN/RAVENparser.py

```diff
@@ -0,0 +1,295 @@
+          else:
+            self.slaveInputFiles.append(os.path.expanduser(moduleToLoad))
+        else:
+          raise IOError(self.printTag+' ERROR: ExternalModel "'+extModel.attrib['name']+'" does not have any attribute named "ModuleToLoad"!!')
+
+    externalFunctions = self.tree.findall('.//Functions/External')
+    if len(externalFunctions) > 0:
+      for extFunct in externalFunctions:
+        if 'file' in extFunct.attrib:
+          moduleToLoad = extFunct.attrib['file']
+          if not moduleToLoad.endswith("py"):
+            moduleToLoad += ".py"
+          if self.workingDir not in moduleToLoad:
+            self.slaveInputFiles.append(os.path.expanduser(os.path.join(self.workingDir,moduleToLoad)))
+          else:
+            self.slaveInputFiles.append(os.path.expanduser(moduleToLoad))
+        else:
+          raise IOError(self.printTag+' ERROR: Functions/External ' +extFunct.attrib['name']+ ' does not have any attribute named "file"!!')
+
+  def returnOutstreamsNamesAnType(self):
+
+    """
+      Method to return the Outstreams names and linked DataObject name
+      @ In, None
+      @ Out, outStreamsNames, dict, the dictionary of outstreams of type print {'outStreamName':[DataObjectName,DataObjectType]}
+    """
+    return self.outStreamsNames
+
+
+  def copySlaveFiles(self,currentDirName):
+    """
+      Method to copy the slave input files
+      @ In, currentDirName, str, the current directory (destination of the copy procedure)
+      @ Out, None
+    """
+    # the dirName is actually in workingDir/StepName/prefix => we need to go back 2 dirs
+    dirName = os.path.join(currentDirName,"../../")
+    # copy SLAVE raven files in case they are needed
+    for slaveInput in self.slaveInputFiles:
+      # full path
+      slaveInputFullPath = os.path.abspath(os.path.join(dirName,slaveInput))
+      # check if exists
+      if os.path.exists(slaveInputFullPath):
+        slaveInputBaseDir = os.path.dirname(slaveInput)
+        slaveDir = os.path.join(currentDirName,slaveInputBaseDir.replace(currentDirName,""))
+        if not os.path.exists(slaveDir):
+          os.makedirs(slaveDir)
+        shutil.copy(slaveInputFullPath,slaveDir)
+      else:
+        raise IOError(self.printTag+' ERROR: File "' +slaveInputFullPath+'" has not been found!!!')
+
+
+  def printInput(self,rootToPrint,outfile=None):
+    """
+      Method to print out the new input
+      @ In, rootToPrint, xml.etree.ElementTree.Element, the Element containing the input that needs to be printed out
+      @ In, outfile, string, optional, output file root
+      @ Out, None
+    """
+    xmlObj = xml.dom.minidom.parseString(ET.tostring(rootToPrint))
+    inputAsString = xmlObj.toprettyxml()
+    inputAsString = "".join([s for s in inputAsString.strip().splitlines(True) if s.strip()])
+    if outfile==None:
+      outfile =self.inputfile
+    IOfile = open(outfile,'wb')
```

Comment:
Is there a reason this is opened as binary 'wb' as opposed to 'w'?

Reply:
No. No reason. I converted it into "w+"